### ISSUE TITLE
Import-AzureRmDataLakeStoreItem command does not work with destination '/' (root) as described in the docs.

### ISSUE THREAD

**Issue by harre096** [Azure/azure-powershell#9055]
## Description
[Import-AzureRmDataLakeStoreItem](https://docs.microsoft.com/en-us/powershell/module/azurerm.datalakestore/import-azurermdatalakestoreitem?view=azurermps-6.13.0) command does not work with destination '/' (root) as described in the docs.

> -Destination
> Specifies the Data Lake Store path to which to upload a file or folder, starting with the root directory (/).

## Steps to reproduce

```powershell
$DataLakeStore = <some dls>
$SourceFilePath = C:/temp/test.txt 
$DestinationFilePath  = /

Import-AzureRmDataLakeStoreItem -AccountName $DataLakeStore -Path $SourceFilePath -Destination $DestinationFilePath -Concurrency 4 -Force -Confirm

#results in 
FailedTransfer:  Path is null
/
```

## Module versions

```powershell
    Directory: C:\Modules\azurerm_6.7.0

ModuleType Version    Name                                ExportedCommands                                             
---------- -------    ----                                ----------------                                             
Script     6.7.0      AzureRM                                                                                          
Script     0.6.12     AzureRM.AnalysisServices            {Resume-AzureRmAnalysisServicesServer, Suspend-AzureRmAnal...
Script     6.1.3      AzureRM.ApiManagement               {Add-AzureRmApiManagementRegion, Get-AzureRmApiManagementS...
Script     0.1.6      AzureRM.ApplicationInsights         {Get-AzureRmApplicationInsights, New-AzureRmApplicationIns...
Script     5.0.4      AzureRM.Automation                  {Get-AzureRMAutomationHybridWorkerGroup, Get-AzureRmAutoma...
Script     4.0.8      AzureRM.Backup                      {Backup-AzureRmBackupItem, Enable-AzureRmBackupContainerRe...
Script     4.1.3      AzureRM.Batch                       {Remove-AzureRmBatchAccount, Get-AzureRmBatchAccount, Get-...
Script     0.14.5     AzureRM.Billing                     {Get-AzureRmBillingInvoice, Get-AzureRmBillingPeriod, Get-...
Script     5.0.3      AzureRM.Cdn                         {Get-AzureRmCdnProfile, Get-AzureRmCdnProfileSsoUrl, New-A...
Script     0.9.8      AzureRM.CognitiveServices           {Get-AzureRmCognitiveServicesAccount, Get-AzureRmCognitive...
Script     5.5.0      AzureRM.Compute                     {Remove-AzureRmAvailabilitySet, Get-AzureRmAvailabilitySet...
Script     0.3.7      AzureRM.Consumption                 {Get-AzureRmConsumptionBudget, Get-AzureRmConsumptionMarke...
Script     0.2.8      AzureRM.ContainerInstance           {New-AzureRmContainerGroup, Get-AzureRmContainerGroup, Rem...
Script     1.0.8      AzureRM.ContainerRegistry           {New-AzureRmContainerRegistry, Get-AzureRmContainerRegistr...
Script     5.0.3      AzureRM.DataFactories               {Remove-AzureRmDataFactory, Get-AzureRmDataFactoryRun, Get...
Script     0.5.8      AzureRM.DataFactoryV2               {Set-AzureRmDataFactoryV2, Update-AzureRmDataFactoryV2, Ge...
Script     5.1.2      AzureRM.DataLakeAnalytics           {Get-AzureRmDataLakeAnalyticsDataSource, New-AzureRmDataLa...
Script     6.1.2      AzureRM.DataLakeStore               {Get-AzureRmDataLakeStoreTrustedIdProvider, Remove-AzureRm...
Script     4.0.7      AzureRM.DevTestLabs                 {Get-AzureRmDtlAllowedVMSizesPolicy, Get-AzureRmDtlAutoShu...
Script     5.0.2      AzureRM.Dns                         {Get-AzureRmDnsRecordSet, New-AzureRmDnsRecordConfig, Remo...
Script     0.3.7      AzureRM.EventGrid                   {New-AzureRmEventGridTopic, Get-AzureRmEventGridTopic, Set...
Script     0.6.8      AzureRM.EventHub                    {New-AzureRmEventHubNamespace, Get-AzureRmEventHubNamespac...
Script     4.1.6      AzureRM.HDInsight                   {Get-AzureRmHDInsightJob, New-AzureRmHDInsightSqoopJobDefi...
Script     5.1.1      AzureRM.Insights                    {Get-AzureRmMetricDefinition, Get-AzureRmMetric, Remove-Az...
Script     3.1.6      AzureRM.IotHub                      {Add-AzureRmIotHubKey, Get-AzureRmIotHubEventHubConsumerGr...
Script     5.1.1      AzureRM.KeyVault                    {Add-AzureKeyVaultCertificate, Update-AzureKeyVaultCertifi...
Script     4.1.2      AzureRM.LogicApp                    {Get-AzureRmIntegrationAccountAgreement, Get-AzureRmIntegr...
Script     0.18.3     AzureRM.MachineLearning             {Move-AzureRmMlCommitmentAssociation, Get-AzureRmMlCommitm...
Script     0.4.6      AzureRM.MachineLearningCompute      {Get-AzureRmMlOpCluster, Get-AzureRmMlOpClusterKey, Test-A...
Script     0.2.5      AzureRM.MarketplaceOrdering         {Get-AzureRmMarketplaceTerms, Set-AzureRmMarketplaceTerms}   
Script     0.10.2     AzureRM.Media                       {Sync-AzureRmMediaServiceStorageKeys, Set-AzureRmMediaServ...
Script     6.5.0      AzureRM.Network                     {Add-AzureRmApplicationGatewayAuthenticationCertificate, G...
Script     5.0.3      AzureRM.NotificationHubs            {Get-AzureRmNotificationHub, Get-AzureRmNotificationHubAut...
Script     5.0.4      AzureRM.OperationalInsights         {New-AzureRmOperationalInsightsAzureActivityLogDataSource,...
Script     1.0.4      AzureRM.PolicyInsights              {Get-AzureRmPolicyEvent, Get-AzureRmPolicyState, Get-Azure...
Script     4.1.8      AzureRM.PowerBIEmbedded             {Remove-AzureRmPowerBIWorkspaceCollection, Get-AzureRmPowe...
Script     5.4.0      AzureRM.profile                     {Disable-AzureRmDataCollection, Disable-AzureRmContextAuto...
Script     4.1.6      AzureRM.RecoveryServices            {Get-AzureRmRecoveryServicesBackupProperty, Get-AzureRmRec...
Script     4.4.0      AzureRM.RecoveryServices.Backup     {Backup-AzureRmRecoveryServicesBackupItem, Get-AzureRmReco...
Script     0.2.8      AzureRM.RecoveryServices.SiteRec... {Edit-AzureRmRecoveryServicesAsrRecoveryPlan, Get-AzureRmR...
Script     5.0.3      AzureRM.RedisCache                  {Remove-AzureRmRedisCachePatchSchedule, New-AzureRmRedisCa...
Script     0.3.8      AzureRM.Relay                       {New-AzureRmRelayNamespace, Get-AzureRmRelayNamespace, Set...
Script     6.4.0      AzureRM.Resources                   {Get-AzureRmProviderOperation, Remove-AzureRmRoleAssignmen...
Script     0.16.8     AzureRM.Scheduler                   {Disable-AzureRmSchedulerJobCollection, Enable-AzureRmSche...
Script     0.6.9      AzureRM.ServiceBus                  {New-AzureRmServiceBusNamespace, Get-AzureRmServiceBusName...
Script     0.3.10     AzureRM.ServiceFabric               {Add-AzureRmServiceFabricApplicationCertificate, Add-Azure...
Script     4.11.1     AzureRM.Sql                         {Get-AzureRmSqlDatabaseTransparentDataEncryption, Get-Azur...
Script     5.0.2      AzureRM.Storage                     {Get-AzureRmStorageAccount, Get-AzureRmStorageAccountKey, ...
Script     4.0.8      AzureRM.StreamAnalytics             {Get-AzureRmStreamAnalyticsFunction, Get-AzureRmStreamAnal...
Script     4.0.5      AzureRM.Tags                        {Remove-AzureRmTag, Get-AzureRmTag, New-AzureRmTag}          
Script     4.0.9      AzureRM.TrafficManager              {Disable-AzureRmTrafficManagerEndpoint, Enable-AzureRmTraf...
Script     4.0.5      AzureRM.UsageAggregates             Get-UsageAggregates                                          
Script     5.0.7      AzureRM.Websites                    {Get-AzureRmAppServicePlan, Set-AzureRmAppServicePlan, New...


    Directory: C:\Modules\azure_5.1.1


ModuleType Version    Name                                ExportedCommands                                             
---------- -------    ----                                ----------------                                             
Script     5.8.2      AzureRM.profile                     {Disable-AzureRmDataCollection, Disable-AzureRmContextAuto...


    Directory: C:\Program Files\WindowsPowerShell\Modules


ModuleType Version    Name                                ExportedCommands                                             
---------- -------    ----                                ----------------                                             
Script     2.1.0      AzureRM                                                                                          
Script     2.1.0      AzureRM.ApiManagement               {Add-AzureRmApiManagementRegion, Get-AzureRmApiManagementS...
Script     2.1.0      AzureRM.Automation                  {Get-AzureRMAutomationHybridWorkerGroup, Get-AzureRmAutoma...
Script     2.1.0      AzureRM.Backup                      {Backup-AzureRmBackupItem, Enable-AzureRmBackupContainerRe...
Script     2.1.0      AzureRM.Batch                       {Remove-AzureRmBatchAccount, Get-AzureRmBatchAccount, Get-...
Script     2.1.0      AzureRM.Cdn                         {Get-AzureRmCdnCustomDomain, New-AzureRmCdnCustomDomain, R...
Script     0.3.0      AzureRM.CognitiveServices           {Get-AzureRmCognitiveServicesAccount, Get-AzureRmCognitive...
Script     2.1.0      AzureRM.Compute                     {Remove-AzureRmAvailabilitySet, Get-AzureRmAvailabilitySet...
Script     2.1.0      AzureRM.DataFactories               {Remove-AzureRmDataFactory, Get-AzureRmDataFactoryRun, Get...
Script     2.1.0      AzureRM.DataLakeAnalytics           {Get-AzureRmDataLakeAnalyticsDataSource, Remove-AzureRmDat...
Script     2.1.0      AzureRM.DataLakeStore               {Add-AzureRmDataLakeStoreItemContent, Export-AzureRmDataLa...
Script     2.1.0      AzureRM.DevTestLabs                 {Get-AzureRmDtlAllowedVMSizesPolicy, Get-AzureRmDtlAutoShu...
Script     2.1.0      AzureRM.Dns                         {Get-AzureRmDnsRecordSet, New-AzureRmDnsRecordConfig, Remo...
Script     2.1.0      AzureRM.HDInsight                   {Get-AzureRmHDInsightJob, New-AzureRmHDInsightSqoopJobDefi...
Script     2.1.0      AzureRM.Insights                    {Add-AzureRmMetricAlertRule, Add-AzureRmLogAlertRule, Add-...
Script     2.1.0      AzureRM.KeyVault                    {Add-AzureKeyVaultCertificate, Set-AzureKeyVaultCertificat...
Script     2.1.0      AzureRM.LogicApp                    {Get-AzureRmIntegrationAccountAgreement, Get-AzureRmIntegr...
Script     0.10.0     AzureRM.MachineLearning             {Export-AzureRmMlWebService, Get-AzureRmMlWebServiceKeys, ...
Script     0.2.0      AzureRM.Media                       {Get-AzureRmMediaServiceNameAvailability, New-AzureRmMedia...
Script     2.1.0      AzureRM.Network                     {Add-AzureRmApplicationGatewayAuthenticationCertificate, G...
Script     2.1.0      AzureRM.NotificationHubs            {Get-AzureRmNotificationHubsNamespaceAuthorizationRules, G...
Script     2.1.0      AzureRM.OperationalInsights         {New-AzureRmOperationalInsightsCustomLogDataSource, Disabl...
Script     2.1.0      AzureRM.PowerBIEmbedded             {Remove-AzureRmPowerBIWorkspaceCollection, Get-AzureRmPowe...
Script     2.1.0      AzureRM.profile                     {Enable-AzureRmDataCollection, Disable-AzureRmDataCollecti...
Script     2.1.0      AzureRM.RecoveryServices            {Get-AzureRmRecoveryServicesBackupProperties, Get-AzureRmR...
Script     2.1.0      AzureRM.RecoveryServices.Backup     {Backup-AzureRmRecoveryServicesBackupItem, Get-AzureRmReco...
Script     2.1.0      AzureRM.RedisCache                  {Remove-AzureRmRedisCachePatchSchedule, New-AzureRmRedisCa...
Script     3.1.0      AzureRM.Resources                   {Get-AzureRmADAppCredential, Get-AzureRmADApplication, Get...
Script     0.10.0     AzureRM.Scheduler                   {Disable-AzureRmSchedulerJobCollection, Enable-AzureRmSche...
Script     2.1.0      AzureRM.ServerManagement            {Install-AzureRmServerManagementGatewayProfile, Reset-Azur...
Script     2.1.0      AzureRM.SiteRecovery                {Stop-AzureRmSiteRecoveryJob, Get-AzureRmSiteRecoveryNetwo...
Script     2.1.0      AzureRM.Sql                         {Get-AzureRmSqlDatabaseRecommendedAction, Get-AzureRmSqlEl...
Script     2.1.0      AzureRM.Storage                     {Get-AzureRmStorageAccount, Get-AzureRmStorageAccountKey, ...
Script     2.1.0      AzureRM.StreamAnalytics             {Get-AzureRmStreamAnalyticsFunction, Get-AzureRmStreamAnal...
Script     2.1.0      AzureRM.Tags                        {Remove-AzureRmTag, Get-AzureRmTag, New-AzureRmTag}          
Script     2.1.0      AzureRM.TrafficManager              {Disable-AzureRmTrafficManagerEndpoint, Enable-AzureRmTraf...
Script     2.1.0      AzureRM.UsageAggregates             Get-UsageAggregates                                          
Script     2.1.0      AzureRM.Websites                    {Get-AzureRmAppServicePlanMetrics, New-AzureRmWebAppDataba...

```

## Debug output

```
DEBUG: 7:24:58 PM - ImportAzureDataLakeStoreItem begin processing with ParameterSet 'NoDiagnosticLogging'.
DEBUG: 7:24:58 PM - using account id '<myid>'...
FailedTransfer:  Path is null
/
DEBUG: AzureQoSEvent: CommandName - Import-AzureRmDataLakeStoreItem; IsSuccess - True; Duration - 00:00:00.2612339; Exception - ;
DEBUG: Finish sending metric.
DEBUG: 7:24:59 PM - ImportAzureDataLakeStoreItem end processing.
DEBUG: 7:24:59 PM - ImportAzureDataLakeStoreItem end processing.
```

## Error output

```

```

**Comment by markcowl** [comment]
@harre096 There are 3 different versions of the Azure cmdlets installed in your PSModulePath.  Can you remove the older versions from your path to ensure that this issue occurs with the latest version?

**Comment by harre096** [comment]
Umm, not really. I am using [Azure DevOps Azure Powershell Step](https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/deploy/azure-powershell?view=azure-devops) so I don't really have control of the environment. This leaves us dependent on Microsoft to keep their agents/tools updated... There is a preview version which supports the newer Az library, but again, this is in preview. 

Regardless, any of the AzureRm versions seem to display this issue. Locally I am using 6.13.0 and I see the same issue: errors and exceptions are being written as terminating errors.

**Comment by markcowl** [comment]
@harre096 Please provide the requested details for your local repro using the latest version.

- Debug stream log ($DebugPreference="Continue" before executing the cmdlet
- Module list (Get-Module -ListAvailable)
- Result of running Resolve-AzureRmError after the error

**Comment by markcowl** [comment]
@harre096 Ping?  We can't proceed without the requested info

**Comment by harre096** [comment]
Sorry Mark.  I have a workaround, so I have unfortunately moved on. If you confirm that this issue does not exist in whatever newer version or AzureRm/Az you happen to be running, that is as much as I was hoping for at this point. :)

As far as I can tell, this bug affects multiple versions before 6.13.0. You already know what version I am running and if you don't want to take the time to chase this on your own, I guess you can cancel/close this issue.

**Comment by rahuldutta90** [comment]
@harre096 yes I agree there is an issue with inputting root as the destination. I will take a look at it and will try to address this in our sdk and release it as part of az release.

**Comment by harre096** [comment]
Thank you guys. Sorry for being a pain and not following your processes in this case.  I appreciate you work and desire to actually fix bugs such as these!

**Comment by rahuldutta90** [comment]
@harre096 Actually i realized that the issue is fixed in 6.2.1 (https://www.powershellgallery.com/packages/AzureRM.DataLakeStore/6.2.1 ). Can you please try this on that version? 

There was a separate issue introduced with 6.2.1. You can mitigate that by passing the full account name with domain

**Comment by markcowl** [comment]
Closing, as sounds like this is resolved.  Please re-open if you see this issue again.
